cdap-common/src/main/java/io/cdap/cdap/common/conf/Constants.java

```diff
@@ -2033,4 +2033,13 @@ public static final class ArtifactCache {
     public static final String BOSS_THREADS = "artifact.cache.boss.threads";
     public static final String WORKER_THREADS = "artifact.cache.worker.threads";
   }
+
+  /**
+   * Constants for MetadataConsumer
+   */
+  public static final class MetadataConsumer {
```

Comment:
I think since you have the context to give you the properties, you can do define these in your SPI implementation, check https://github.com/cdapio/cdap/blob/develop/cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/provisioner/ProvisionerContext.java#L48-L57 for example on how to define these.